Package.swift

```diff
@@ -110,12 +102,20 @@ package.targets.append(contentsOf: [
 ])
 #endif
 
+/// Whether this package is building for distribution.
+///
+/// This can be used to conditionalize certain settings which are not intended
+/// for use when building the package for distribution to end users.
+/// Non-distribution (i.e. development-only) builds may enable things
+/// like stricter compiler features which are unsupported or inappropriate for
+/// client builds.
+let isBuildingForDistribution = false
```

Comment:
What if instead we checked for an environment variable and built SPI when it's set?

```swift
let buildSPIDocumentation: Bool = {
  if let boolValue = Context.environment["SWT_SPI_DOCUMENTATION_ENABLED"].flatMap(Bool.init) {
    return boolValue
  } else if let intValue = Context.environment["SWT_SPI_DOCUMENTATION_ENABLED"].flatMap(UInt64.init) {
    return intValue != 0
  }
  return false // disabled by default
}()
```

(That's a slightly simplified version of what our own `Environment` type does.)

Reply:
We could do this, although a workflow I commonly use for viewing docs is the "Product > Build Documentation" command in recent versions of Xcode, and I was hoping for that to get this flag too. I don't know of a way to apply an env var in that scenario (when using the GUI Xcode app, not its xcodebuild CLI).

Reply:
The constant I added here would still be needed, or at least it would still be helpful, even if we conditionalized the SPI docs flag based on an env var, because the constant is what (now) decides whether all the unsafeFlags get included

Reply:
Honestly we can also just drop the unsafe flags outright.

Reply:
> although a workflow I commonly use for viewing docs is the "Product > Build Documentation" command in recent versions of Xcode

We can ask the DocC folks if they can help with this.

Reply:
> We could do this, although a workflow I commonly use for viewing docs is the "Product > Build Documentation" command in recent versions of Xcode, and I was hoping for that to get this flag too. I don't know of a way to apply an env var in that scenario (when using the GUI Xcode app, not its xcodebuild CLI).

+1 for some similar scenario I experienced. And I believe it is a Xcode/SwiftPM issue instead of a DocC issue.

Xcode only gives us the ability to set env value for executable target when they are running. I could not find a easy solution to setup the env value I use in my Package.swift file.

Currently I just use a small macOS [app](https://github.com/Kyle-Ye/MenuHelper) I wrote to trigger `export XX=YY && open ./ -a Xocde` for me in Finder.

Otherwise we'll have to quit Xcode and manually export the env value in shell and then open Xcode in the shell to make Package.swift access the env we set.

<img width="612" alt="image" src="https://github.com/apple/swift-testing/assets/43724855/d2a89a39-e5bd-4f5d-bba0-cff6d523fc4a">

As a person who maintains some Swift packages and likes to add environment variable condition config in package, it is really troublesome at present if we do not use any other 3rd party tools.